WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -298,16 +299,45 @@ def edit
     render :edit
   end
 
+  # TODO: Refactor for cleaner integration of Stripe and Paypal (and other payment services)
   def payment_setup
     @competition = competition_from_params(includes: CHECK_SCHEDULE_ASSOCIATIONS)
 
+    # Stripe setup URL
     client = create_stripe_oauth_client
     oauth_params = {
       scope: 'read_write',
       redirect_uri: EnvConfig.ROOT_URL + competitions_stripe_connect_path,
       state: @competition.id,
     }
     @authorize_url = client.auth_code.authorize_url(oauth_params)
+
+    # Paypal setup URL
+    # TODO: Don't generate this URL if there is already a connected paypal account?
+    @paypal_onboarding_url = PaypalInterface.generate_paypal_onboarding_link(@competition.id) unless Rails.env.production?
+  end
+
+  def paypal_return
+    unless Rails.env.production?
```

Comment:
I feel like a better (as in, a little more human-friendly) approach would be
```ruby
def paypal_return
  if Rails.env.production?
    flash[:error] = 'PayPal is not yet available in production environments' # No need to properly i18n this, as it's just an interim dev message
    return redirect_to root_url # or whatever other URL you feel makes sense here
  end
  
  # ... execute the method as normal without wrapping the _whole_ thing into a giant if/unless
end
```

Reply:
Absolutely wow yes this is miles better